File: U1/EXAMEN/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/Models/Cliente.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BanquitoServer_Soap_DotNet_GR01.Models
{
    /// <summary>
    /// Entidad Cliente del banco
    /// </summary>
    [Table("Cliente")]
    public class Cliente
    {
        [Key]
        [Column("ClienteId")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long ClienteId { get; set; }

        [Required]
        [Column("Cedula")]
        [MaxLength(10)]
        [Index("IX_Cliente_Cedula", IsUnique = true)]
        public string Cedula { get; set; }

        [Required]
        [Column("Nombres")]
        [MaxLength(100)]
        public string Nombres { get; set; }

        [Required]
        [Column("Apellidos")]
        [MaxLength(100)]
        public string Apellidos { get; set; }

        [Required]
        [Column("FechaNacimiento")]
        public DateTime FechaNacimiento { get; set; }

        [Column("EstadoCivil")]
        [MaxLength(20)]
        public string EstadoCivil { get; set; }

        [Column("Direccion")]
        [MaxLength(200)]
        public string Direccion { get; set; }

        [Column("Telefono")]
        [MaxLength(20)]
        public string Telefono { get; set; }

        [Column("Email")]
        [MaxLength(100)]
        public string Email { get; set; }

        // Propiedades de navegación
        public virtual ICollection<Cuenta> Cuentas { get; set; }
        public virtual ICollection<Credito> Creditos { get; set; }

        // Propiedad calculada
        [NotMapped]
        public string NombreCompleto
        {
            get { return $"{Nombres} {Apellidos}"; }
        }

        public Cliente()
        {
            Cuentas = new HashSet<Cuenta>();
            Creditos = new HashSet<Credito>();
        }
    }
}

File: U2/PAO202551_ELECTRO_BDDNUBE_RESTFULL_JAVA_GR01/04. SERVIDOR/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Models/Cliente.java
package ec.edu.monster.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

/**
 * Entidad Cliente del banco
 */
@Entity
@Table(name = "Cliente")
@Data
@NoArgsConstructor
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ClienteId")
    private Long clienteId;

    @Column(name = "Cedula", nullable = false, length = 10, unique = true)
    private String cedula;

    @Column(name = "Nombres", nullable = false, length = 100)
    private String nombres;

    @Column(name = "Apellidos", nullable = false, length = 100)
    private String apellidos;

    @Column(name = "FechaNacimiento", nullable = false)
    private LocalDate fechaNacimiento;

    @Column(name = "EstadoCivil", length = 20)
    private String estadoCivil;

    @Column(name = "Direccion", length = 200)
    private String direccion;

    @Column(name = "Telefono", length = 20)
    private String telefono;

    @Column(name = "Email", length = 100)
    private String email;

    // Relaciones
    @OneToMany(mappedBy = "cliente", fetch = FetchType.LAZY)
    private Set<Cuenta> cuentas = new HashSet<>();

    @OneToMany(mappedBy = "cliente", fetch = FetchType.LAZY)
    private Set<Credito> creditos = new HashSet<>();

    // Propiedad calculada
    @Transient
    public String getNombreCompleto() {
        return nombres + " " + apellidos;
    }
}
